www/pages/plan-trip/plan-trip-controller.js

```diff
@@ -156,121 +162,144 @@ angular.module('pvta.controllers').controller('PlanTripController', function ($s
 
     var originAutocomplete = new google.maps.places.Autocomplete(originInput);
     var destinationAutocomplete = new google.maps.places.Autocomplete(destinationInput);
-    originAutocomplete.setBounds($scope.bounds);
-    destinationAutocomplete.setBounds($scope.bounds);
+    originAutocomplete.setBounds(bounds);
+    destinationAutocomplete.setBounds(bounds);
 
     originAutocomplete.addListener('place_changed', function () {
-      $scope.params.destinationOnly = false;
-      var place = originAutocomplete.getPlace();
-      if (!place.geometry) {
-        console.error('Place has no geometry.');
-        return;
-      }
-      if ($scope.bounds.contains(place.geometry.location)) {
-        expandViewportToFitPlace($scope.map, place);
-        $scope.params.origin.id = place.place_id;
-        $scope.params.origin.name = place.name;
+      mapLocation(originAutocomplete.getPlace(), function (place) {
+        $scope.request.origin.id = place.place_id;
+        $scope.request.origin.name = place.name;
+        //Name the trip if there is a destination: ORIGIN to DESTINATION
+        if ($scope.request.destination.name) {
+          $scope.request.name = place.name + ' to ' + $scope.request.destination.name;
+        }
         $scope.$apply();
-      } else {
-        $scope.params.origin.id = null;
-        originInput.value = $scope.params.origin.name;
-        invalidLocationPopup();
-      }
+      }, function (err) {
```

Comment:
Since we take the time to handle the error case, might as well add some analytics to it.  An Analytics event similar to [this one](https://github.com/umts/pvta-multiplatform/blob/master/www/factories/map-factories.js#L30) might do the trick.